api/cluster_create.go

```diff
@@ -149,7 +149,7 @@ func (a *ClusterAPI) CreateCluster(
 		PostHooks:      postHooks,
 	}
 
-	creator := cluster.NewCommonClusterCreator(createClusterRequest, commonCluster)
+	creator := cluster.NewClusterCreator(createClusterRequest, commonCluster, a.tokenGenerator, a.externalBaseURL)
```

Comment:
Let's create a separate function for the common and pke cluster creator and conditionally call them here.

Reply:
why do you think that it's the api's responsibility to decide on how a specific type of clusters should be created?

Reply:
It's not, but for now it's an acceptable compromise, until we figure out how to properly do this. Once we have a few more creators, having a single constructor will become unmaintainable. Refactoring a conditional logic is much easier.